include/revng/PipeboxCommon/Helpers/PipeRunner.h

```diff
@@ -21,39 +21,34 @@ namespace revng::pypeline::helpers {
 template<typename ContainerListUnwrapper>
 struct PipeRunner {
 private:
-  template<size_t... I>
-  using integer_sequence = std::integer_sequence<size_t, I...>;
   using ObjectDeps = pypeline::ObjectDependencies;
 
   using ListType = ContainerListUnwrapper::ListType;
 
 public:
-  template<IsPipe T, typename... ContainersT>
+  template<IsPipe T>
   static ObjectDeps run(T &Pipe,
-                        ObjectDeps (T::*RunMethod)(const Model &,
-                                                   const pypeline::Request &,
-                                                   const pypeline::Request &,
-                                                   llvm::StringRef,
-                                                   ContainersT...),
                         const Model &TheModel,
                         const pypeline::Request &Incoming,
                         const pypeline::Request &Outgoing,
                         llvm::StringRef Configuration,
                         ListType Containers) {
-    revng_assert(Incoming.size() == sizeof...(ContainersT));
-    revng_assert(Outgoing.size() == sizeof...(ContainersT));
+    using Traits = PipeRunTraits<T>;
```

Comment:
It's a "trait", not "trait*s*".
Fix everywhere.

Reply:
I've always seen the plural, e.g. the `llvm::yaml::*Traits` classes for serialization, `llvm::GraphTraits`, `KeyedObjectTraits` etc.